rclcpp_lifecycle/test/test_lifecycle_service_client.cpp

```diff
@@ -355,3 +356,55 @@ TEST_F(TestLifecycleServiceClient, lifecycle_transitions) {
   transitions = lifecycle_client()->get_available_transitions();
   EXPECT_EQ(transitions.size(), 0u);
 }
+
+
+TEST_F(TestLifecycleServiceClient, get_service_names_and_types_by_node)
+{
+  auto node1 = std::make_shared<LifecycleServiceClient>("client1");
+  auto node2 = std::make_shared<LifecycleServiceClient>("client2");
+
+  const auto * node_graph =
+    dynamic_cast<rclcpp::node_interfaces::NodeGraph *>(node1->get_node_graph_interface().get());
+  ASSERT_NE(nullptr, node_graph);
+
+  EXPECT_THROW(
+    node_graph->get_service_names_and_types_by_node("not_a_node", "not_absolute_namespace"),
+    std::runtime_error);
+  auto service_names_and_types1 = node_graph->get_service_names_and_types_by_node("client1", "/");
+  auto service_names_and_types2 = node_graph->get_service_names_and_types_by_node("client2", "/");
+  EXPECT_EQ(service_names_and_types1.size(), service_names_and_types2.size());
+}
+
+TEST_F(TestLifecycleServiceClient, declare_parameter_with_no_initial_values)
+{
+  auto node1 = std::make_shared<LifecycleServiceClient>("client1");
+
+  auto on_set_parameters =
+    [](const std::vector<rclcpp::Parameter> & parameters) {
```

Comment:
nit: if you don't need to use `parameters` in this lambda, you can just omit the name of the argument so you don't have to cast to void.
```suggestion
    [](const std::vector<rclcpp::Parameter> &) {
```